TYPO3.Flow/Classes/TYPO3/Flow/Security/Context.php

```diff
@@ -837,7 +864,18 @@ public function getContextHash()
             $this->initialize();
         }
         if ($this->contextHash === null) {
-            $this->contextHash = md5(implode('|', array_keys($this->getRoles())));
+            $contextHashSoFar = implode('|', array_keys($this->getRoles()));
+
+            $this->withoutAuthorizationChecks(function () use (&$contextHashSoFar) {
+                foreach ($this->globalObjects as $globalObjectsRegisteredClassName) {
+                    $globalObject = $this->objectManager->get($globalObjectsRegisteredClassName);
```

Comment:
Yep, I thought you pointed at `instanceof` not returning correct result because the object might be a proxy, but it's just about avoiding to load the object at all :)